### PR TITLE
Add Loading indicator and Use async storage to store to retreive todos

### PR DESCRIPTION
**Pull request by jawakarD**
image

Todo Stored ans retrived from AsyncStorage
image

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -28,8 +28,15 @@ import AddTodo from 'components/Todo/AddTodo';
 const store = configureStore();
 sagaMiddleWare.run(rootSaga);
 
+export type ModalStackParamsList = {
+  Main: undefined;
```
Comment (revathskumar) [inline App.tsx:32]: Why `undefined` in type?

File: src/components/Todo/AddTodo.tsx [hunk 2]
```diff
@@ -1,14 +1,53 @@
...
-    </View>
+    <ScrollView style={styles.addTodoModal}>
+      <TextInput
+        style={[styles.todoInput, styles.text]}
+        onChangeText={(text) => setTodo(text)}
+        placeholder="Add todo"
+        value={todo}
+      />
+      <View>
+        <TouchableOpacity onPress={() => addTodo()} style={styles.submitButton}>
```
Comment (revathskumar) [inline src/components/Todo/AddTodo.tsx:39]: `onPress={addTodo}`

File: src/package.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (revathskumar) [inline src/package.json:1]: Why this?

File: src/components/Todo/types.d.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import {StackNavigationProp} from '@react-navigation/stack';
+import {TodoStackParamsList} from './TodoNavigator';
+import {ModalStackParamsList} from '../../../App';
```
Comment (revathskumar) [inline src/components/Todo/types.d.ts:3]: can we use `src/App` instead of `../../../App`?

File: src/reducers/todo.ts [hunk 5]
```diff
@@ -38,6 +38,12 @@ const todoReducer = (state = initialState, action: any) => {
         ),
       };
 
+    case ADD_TODO:
+      return {
+        ...state,
+        todos: [...state.todos, action.todo],
```
Comment (revathskumar) [inline src/reducers/todo.ts:44]: We can use immer instead it this. (No need to make changes in this PR. )

File: src/sagas/index.js [hunk 6]
```diff
@@ -27,7 +37,40 @@ function* getTodosSaga() {
   try {
     const todos = yield call(getTodos);
 
-    yield put({type: ADD_TODOS, todos: todos.data});
+    if (todos !== null) {
+      yield put({type: ADD_TODOS, todos: JSON.parse(todos).todos});
```
Comment (revathskumar) [inline src/sagas/index.js:41]: Similarly how we have separated `api` in we might need a wrapper for AsyncStorage. and this `JSON.parse` should go into that.

File: src/reducers/todo.ts (hunk 5 above)
Comment (jawakarD) [reply src/reducers/todo.ts:44]: yup 👍

File: src/sagas/index.js (hunk 6 above)
Comment (jawakarD) [reply src/sagas/index.js:41]: Got it 👍

File: src/components/Todo/types.d.ts (hunk 4 above)
Comment (jawakarD) [reply src/components/Todo/types.d.ts:3]: Used this before adding package to `src` folder, will remove.

File: src/package.json (hunk 3 above)
Comment (jawakarD) [reply src/package.json:1]: https://github.com/facebook/react-native/issues/3099#issuecomment-221815006

Any other way to do absolute path in react native?

File: src/components/Todo/AddTodo.tsx (hunk 2 above)
Comment (jawakarD) [reply src/components/Todo/AddTodo.tsx:39]: 👍

File: App.tsx (hunk 1 above)
Comment (jawakarD) [reply App.tsx:32]: https://reactnavigation.org/docs/typescript#type-checking-the-navigator

Typecheck for params.

File: src/package.json (hunk 3 above)
Comment (revathskumar) [reply src/package.json:1]: understood. Not sure about the other way.

File: App.tsx (hunk 1 above)
Comment (revathskumar) [reply App.tsx:32]: will check

**Pull request merged**